src/sample_weight_audit/estimator_check.py

```diff
@@ -178,6 +158,26 @@ def compute_predictions(est, X):
         raise NotImplementedError(f"Estimator type not supported: {est}")
 
 
+def score_estimator(est, X, y=None, score=True):
+
+    preds = compute_predictions(est, X)
+    if not score:
+        return preds
+    if y is None:
+        print("y-values are required when score is set to True")
+    if is_regressor(est):
+        return mean_squared_error(preds, y)
+    elif is_classifier(est):
+        if hasattr(est, "predict_proba"):
+            return log_loss(y, preds)
+        else:
+            return roc_auc_score(preds, y)
+    elif hasattr(est, "transform"):
+        return rand_score(preds, y)
```

Comment:
We should use `rand_score` in estimators that are clusterers (detected with `is_clusterer`), not on all transformers.

```suggestion
    elif is_clusterer(est):
        return rand_score(preds, y)
```

Transformers (that are not clusterers) should instead be pipelined with a ridge regressor and therefore treated as regressors.

Reply:
Similarly, `check_pipeline_and_fit` should be updated to avoid pipelining clusterers.